utils/proxy.py

```diff
@@ -70,3 +74,23 @@ def get_proxy(agentConfig):
                   "Proxy is probably not set", str(e))
 
     return None
+
+
+def config_proxy_skip(proxies, uri, skip_proxy=False):
```

Comment:
Could you add a comment explaining what this function does?

Also, a unit test of this function in `tests/core/test_proxy.py` would be perfect :)